firefox-ios/Client/Frontend/Browser/BrowserViewController/State/BrowserViewControllerState.swift

```diff
@@ -224,9 +229,33 @@ struct BrowserViewControllerState: ScreenState, Equatable {
                 toast: state.toast,
                 windowUUID: state.windowUUID,
                 browserViewType: state.browserViewType,
-                navigateToHome: state.navigateToHome,
+                navigateTo: nil,
                 showQRcodeReader: true,
                 microsurveyState: MicrosurveyPromptState.reducer(state.microsurveyState, action))
+        case GeneralBrowserActionType.navigateBack:
+            return BrowserViewControllerState(
+                searchScreenState: state.searchScreenState,
+                showDataClearanceFlow: state.showDataClearanceFlow,
+                toolbarState: state.toolbarState,
```

Comment:
do we want to add the call to reducer for ToolbarState?

Reply:
It seems the path forward for these states is to call the reducers all the time. This allows more flexibility in that the reducer can determine whether it wants to use existing state or create a new state. This avoids needing the follow the code backwards in the future to determine if its actually being reduced properly, in case the reducer does respond to a different action. 

Currently, the reducers only do logical computations (or can just return the existing state), so should not cause performance issues from calling it all the time.
@PARAIPAN9